src/stream/src/task/barrier_manager/tests.rs

```diff
@@ -171,8 +171,8 @@ async fn test_managed_barrier_collection_before_send_request() -> StreamResult<(
     let barrier = Barrier::new_test_barrier(curr_epoch);
     let epoch = barrier.epoch.prev;
 
-    // Collect a barrier before sending
-    manager.collect(extra_actor_id, &barrier);
+    // Read the mutation after receiving the barrier from remote input.
+    let mutation_reader = pin!(manager.read_barrier_mutation(&barrier));
```

Comment:
Might poll the future with `poll_fn` and assert the the future is pending like in other places in this test.